addons/mail/models/mail_channel.py

```diff
@@ -934,7 +931,7 @@ def channel_seen(self, last_message_id=None):
         kept only for compatibility reasons.
         """
         self.ensure_one()
-        domain = [('channel_ids', 'in', self.ids)]
+        domain = ["&", ("model", "=", "mail.channel"), ("res_id", "in", self.ids)]
```

Comment:
same question

Reply:
Well expression.AND will basically do the same ^^ . For simple domains we can do it manually until we have a better domain management tool. For the explicit & it is not mandatory indeed. I prefer explicit to implicit.